### PR TITLE
generic_login_error_message

### PR DESCRIPTION
**Pull request by anikolskiy**

### CODE REVIEW

File: app/listeners/casino/login_credential_acceptor_listener.rb [hunk 1]
```diff
@@ -4,7 +4,10 @@ class CASino::LoginCredentialAcceptorListener < CASino::Listener
   attr_reader :controller
 
   def user_logged_in(url, ticket_granting_ticket, cookie_expiry_time = nil)
-    @controller.cookies[:tgt] = { value: ticket_granting_ticket, expires: cookie_expiry_time, httponly: true, secure: true }
+    tgt_cookie = { value: ticket_granting_ticket, expires: cookie_expiry_time, httponly: true }
+    tgt_cookie[:secure] = true unless Rails.env.test?
```
Comment (stereosupersonic) [inline app/listeners/casino/login_credential_acceptor_listener.rb:8]: what is the problem within the tests?

File: app/models/casino/service_ticket.rb [hunk 2]
```diff
@@ -7,25 +7,27 @@ class CASino::ServiceTicket < ActiveRecord::Base
...
 
   def self.cleanup_consumed_hard
-    self.delete_all(['created_at < ? AND consumed = ?', (CASino.config.service_ticket[:lifetime_consumed].seconds * 2).ago, true])
+    where('created_at < ? AND consumed = ?', (CASino.config.service_ticket[:lifetime_consumed].seconds * 2).ago, true).delete_all
   end
 
 
   def service=(service)
-    normalized_encoded_service = Addressable::URI.parse(service).normalize.to_str
+    normalized_encoded_service = Addressable::URI.parse(service)&.normalize&.to_str || ''
```
Comment (stereosupersonic) [inline app/models/casino/service_ticket.rb:23]: should be double quotes

File: app/processors/casino/processor_concern/service_tickets.rb [hunk 3]
```diff
@@ -41,7 +41,12 @@ def clean_service_url(dirty_service)
 
         service_uri.path = (service_uri.path || '').gsub(/\/+\z/, '').sub('.html', '') # IE hack
         service_uri.path = '/' if service_uri.path.blank?
-        clean_service = service_uri.normalize.to_s
+
+        clean_service = begin
+                        service_uri.normalize.to_s
```
Comment (stereosupersonic) [inline app/processors/casino/processor_concern/service_tickets.rb:46]: does "normalize" raise the `InvalidURIError`?

**Review by stereosupersonic — APPROVED** [review]
OLGTM

**Comment by stereosupersonic** [comment]
the original deployed branch was "password_expiration_rails_6" does this branch also include these changes?

**Review by ahmedm10 — APPROVED** [review]

File: app/listeners/casino/login_credential_acceptor_listener.rb (hunk 1 above)
Comment (anikolskiy) [reply app/listeners/casino/login_credential_acceptor_listener.rb:8]: they run over http

File: app/models/casino/service_ticket.rb (hunk 2 above)
Comment (anikolskiy) [reply app/models/casino/service_ticket.rb:23]: if we'd enable rubocop here, I think there would be more refactoring to do. Here, I follow the gem's original convention (single quotes).

File: app/processors/casino/processor_concern/service_tickets.rb (hunk 3 above)
Comment (anikolskiy) [reply app/processors/casino/processor_concern/service_tickets.rb:46]: Specs do pass here. But they don't pass on master: https://travis-ci.com/github/konvenit/casino/builds/171597013. So I'd say everything works correctly.

**Comment by anikolskiy** [comment]
@stereosupersonic You merged and deleted "password_expiration_rails_6" branch: https://github.com/konvenit/CASino/commit/ee64b12f91d1a03fab70dfd8a8154f4ef401d298 . But you didn't update casinator's Gemfile: https://github.com/konvenit/casinator/blob/master/Gemfile#L22 . That was a bit confusing for a short while. Now, this PR is based on *master*. And casinator's gemfile fixed here: https://github.com/konvenit/casinator/pull/235/files#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fR22. Once this PR is merged, I'll update casinator to properly use master instead.

File: app/models/casino/service_ticket.rb (hunk 2 above)
Comment (stereosupersonic) [reply app/models/casino/service_ticket.rb:23]: rubocop is just a tool that supervises our coding guidelines, and we decided to use double quotes https://github.com/konvenit/guidelines/blob/master/ruby.md#consistent-string-literals

**Pull request merged**